### DISCUSSION TITLE
Getting the default hover provider to work

### DISCUSSION THREAD

**Discussion by ballcoach12** [eclipse-langium/langium#410] [Q&A]
I'm wondering what I need to do in order to get the default hover provider implementation to work. I suspect that it should just work out of the box when I start up the language server. I notice that it works fine in the VS Code extension for the langium grammar language. 

I am tracing the lsp messages between  the client and server, and I can see that the editor sends  the textDocument/hover  request, but the message is always 'No result returned.' I suspect that either the MultilineCommendHoverProvider is not getting called, or that no  content is being returned from the getAstNodeHoverContent() function. I suspect the former, but I'm not able to set a debug breakpoint in that function; VS Code always says 'unbound breakpoint'.  

In summary, I'm wondering if there is anything special that I need to do to get this to work in  0.2.0. I have a multi-line comment in my grammar for my AST root node, so I would hope that the contents of that comment would be returned.

**Comment by msujew** [comment]
Hi @ballcoach12,

our behavior to identify what a "multiline comment" exactly is differs a bit from Xtext. While Xtext uses convention, i.e. "search for a token named `ML_COMMENT`", we use a heuristic to identify any comments which might be multiline comments. It could be that it does not pick up on your multiline comment definition. In that case, you could simply override the `LangiumServices.parser.GrammarConfig` service to return `{ multilineCommentRules:  ['ML_COMMENT'] }`.

On the unbound breakpoint issue: Are you attaching to your language server process, or only debugging the vscode extension? Since the language server runs in a separate process, you need an additional launch config to debug it. See [this for example](https://github.com/langium/langium/blob/a4c008c0ad8ced9142c77fc62482da0981311f13/.vscode/launch.json#L47-L62).

**Reply by ballcoach12** [reply]
Hi @msujew , thanks for the quick reply.  I'll pursue overriding `LangiumServices.parser.GrammarConfig' as you suggest. I assume that it would also be acceptable to simply turn off the default hover provider and add my own, if all else fails. But since the comment for the respective grammar element is what I really want, I hope I don't have to do that.

As far as the debugging goes, here is my launch.config:

```
{
			"type": "extensionHost",
			"request": "launch",
			"name": "Launch Client",
			"runtimeExecutable": "${execPath}",
			"args": ["--extensionDevelopmentPath=${workspaceRoot}"],
			"outFiles": ["${workspaceRoot}/client/out/**/*.js"],
			"preLaunchTask": {
				"type": "npm",
				"script": "watch"
			}
		},
		{
			"type": "node",
			"request": "attach",
			"name": "Attach to Server",
			"port": 6009,
			"sourceMaps": true,
			"restart": true,
			"outFiles": [
				"${workspaceRoot}/out/**/*.js"
			
			]
		}

```
I am  launching both client and server when I debug.

**Reply by msujew** [reply]
I see. I think adding `${workspaceRoot}/node_modules/langium/**/*.js` to your `outFiles` of the `Attach to Server` config might help this. I'm quite sure hovering runs the code, so VSCode displaying the breakpoints as "unbound" is surely a config issue.
